Azure Architecture guidance (#6140)
Particular/docs.particular.net: architecture/azure/index.md

### Design principles

The Particular Service Platform makes systems follow the [ten design principles for Azure applications](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/).

* [Self healing](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/self-healing) is provided by the rich set of [recoverability](/architecture/recoverability.md) features like automatic retries, load leveling, throttling and more, which make application services resilient to failures.
* [Redundancy](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/redundancy) is provided by capabilities such as [scaling out](/nservicebus/scaling.md#scaling-out-to-multiple-nodes) and [high availability](/nservicebus/scaling.md#high-availability).

> * [Coordination is minimized](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/minimize-coordination) between services by [messaging](/architecture/messaging.md). See [this talk](https://www.youtube.com/watch?v=0TYbHVc2yWI) for more information.

Author: ```suggestion
* [Coordination is minimized](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/minimize-coordination) between services by [messaging](/architecture/messaging.md). Watch [_Autonomous microservices don't share data. Period_ (video)](https://www.youtube.com/watch?v=0TYbHVc2yWI) for more recommendatios.
```

* [Scaling out](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/scale-out) is achieved by various methods such as [asynchronous message-based communication](https://learn.microsoft.com/en-us/dotnet/architecture/microservices/architect-microservice-container-applications/asynchronous-message-based-communication) and [competing consumers](/nservicebus/scaling.md#scaling-out-to-multiple-nodes-competing-consumers).

> * [Partitioning](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/partition) is achieved by decomposing and offloading resource-intensive processes using [workflows](/architecture/workflows.md) and the integration of the Particular Service Platform with technology-specific partitioning capabilities.

Author: good catch. It looks like https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/build-for-business is missing.

* [Managed services](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/managed-services) such as Azure App Service, Azure SQL, and Azure Cosmos DB are fully supported by the Particular Service Platform. See [technology choices](#technology-choices).
* [A wide variety of data stores](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/use-best-data-store) are supported by the Particular Service Platform as [persistence choices](/persistence).
* [System evolution](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/design-for-evolution) is aided by:
  * High cohesion within [service boundaries](https://particular.net/webinars/finding-your-service-boundaries-a-practical-guide).
  * Loose coupling and asynchronous operations through [messaging](/architecture/messaging.md).
  * Encapsulating domain knowledge in [message handlers and sagas](/nservicebus/handlers-and-sagas.md).
  * Separating domain logic from infrastructure concerns, such as reliability and recoverability, which are handled by the Particular Service Platform.

>   * Enabling independent services deployment by decoupling their communication using [asynchronous messaging](/nservicebus/messaging/).

Author: @DavidBoike thoughts about this missing point? (the alternative would be to just leave it out)

```suggestion
  * Enabling independent services deployment by decoupling their communication using [asynchronous messaging](/nservicebus/messaging/).
* [Meet business needs](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/build-for-business) by decomposing workloads into logically separated units using [messaging](/architecture/messaging.md). Allow each component's characteristics and design decisions can be optimized and [monitored](/servicepulse) for the specific business needs at hand (e.g. [resiliency](/architecture/recoverability.md), [scalability](/nservicebus/scaling.md), [consistency](/architecture/consistency.md), etc.).
```

[**Talk to a solution architect and learn more about how the Particular Service Platform helps systems follow Azure design principles. →**](https://particular.net/solution-architect?message=I%27d%20like%20to%20talk%20to%20a%20solution%20architect%20to%20learn%20more%20about%20how%20the%20Particular%20Service%20Platform%20helps%20systems%20follow%20Azure%20design%20principles.)

### Design patterns

[Azure Cloud Design Patterns](https://learn.microsoft.com/en-us/azure/architecture/patterns/) lists many proven patterns for addressing specific challenges. The Particular Service Platform implements many of these such as [Asynchronous Request-Reply](/nservicebus/messaging/reply-to-a-message.md), [Circuit Breaker](/nservicebus/recoverability/#automatic-rate-limiting), [Claim Check](/nservicebus/messaging/databus/), [Competing Consumers](/nservicebus/scaling.md#scaling-out-to-multiple-nodes-competing-consumers), and more.

> [**Talk to a solution architect to help identify suitable design patterns→**](https://particular.net/solution-architect?message=I%27d%20like%20to%20talk%20to%20a%20solution%20architect%20to%20help%20identify%20suitable%20design%20patterns%20for%20my%20system.)

Author: I think dots at the end of ctas (which end with the arrow) feel off?

> [**Talk to a solution architect to help identify suitable design patterns→**](https://particular.net/solution-architect?message=I%27d%20like%20to%20talk%20to%20a%20solution%20architect%20to%20help%20identify%20suitable%20design%20patterns%20for%20my%20system.)

Reviewer: Ah - because of the arrow, I screwed up where I put the dot. This matches what you have in the CTA on line 57.
```suggestion
[**Talk to a solution architect to help identify suitable design patterns. →**](https://particular.net/solution-architect?message=I%27d%20like%20to%20talk%20to%20a%20solution%20architect%20to%20help%20identify%20suitable%20design%20patterns%20for%20my%20system.)
```

> [**Talk to a solution architect to help identify suitable design patterns→**](https://particular.net/solution-architect?message=I%27d%20like%20to%20talk%20to%20a%20solution%20architect%20to%20help%20identify%20suitable%20design%20patterns%20for%20my%20system.)

Author: maybe rather remove the dots on both of them?

> [**Talk to a solution architect to help identify suitable design patterns→**](https://particular.net/solution-architect?message=I%27d%20like%20to%20talk%20to%20a%20solution%20architect%20to%20help%20identify%20suitable%20design%20patterns%20for%20my%20system.)

Author: aligned both of the links and avoided the dots.

* [A wide variety of data stores](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/use-best-data-store) are supported by the Particular Service Platform as [persistence choices](/persistence).
* [System evolution](https://learn.microsoft.com/en-us/azure/architecture/guide/design-principles/design-for-evolution) is aided by:
  * High cohesion within [service boundaries](https://particular.net/webinars/finding-your-service-boundaries-a-practical-guide).
  * Loose coupling and asynchronous operations through [messaging](/architecture/messaging.md)
  * Encapsulating domain knowledge in [message handlers and sagas](/nservicebus/handlers-and-sagas.md).
  * Separating domain logic from infrastructure concerns, such as reliability and recoverability, which are handled by the Particular Service Platform.
  * Deploying services independently through the use of [asynchronous messaging](/nservicebus/messaging/).

> [**Talk to a solution architect and learn more about how the Particular Service Platform helps systems follow Azure design principles. →**](https://particular.net/solution-architect?message=I%27d%20like%20to%20talk%20to%20a%20solution%20architect%20to%20learn%20more%20about%20how%20the%20Particular%20Service%20Platform%20helps%20systems%20follow%20Azure%20design%20principles.)

Author: it's already bold but I agree that the CTAs are a bit subtle. Thinking about some ideas to make them more visible without being completely obnoxious.
